hack/e2e/main.go

```diff
@@ -0,0 +1,94 @@
+package main
```

Comment:
This module does part of test/integration/suite_test.go job  - build/re-use existing GKE cluster, similar as what we did for kind test cluster.

Actionable: integrate this into suite_test.go module.

Reply:
This looks intentional, since the stuff in this section is specific to GKE and, per the comment, will eventually move to KTF. It's about setting up the cluster, whereas suite_test.go is concerned with setting up the controller.

Reply:
currently, suite_test either re-use existing or bump new kind cluster.
https://github.com/Kong/kubernetes-ingress-controller/blob/next/test/integration/suite_test.go#L131
https://github.com/Kong/kubernetes-ingress-controller/blob/next/test/integration/suite_test.go#L137 does exact

Reply:
The integration test suite now has _support_ for GKE, but it does not _create_ GKE clusters. This is intentional: we want to have a discreet E2E test suite in time but for now the best we have is the integration test suite to run against production grade clusters.

I am strongly against adding GKE builds as part of the normal integration test suite: we've been able to maintain a lot of good testing using local cluster implementations with Kind and in general we should continue to focus on that as the standard.

There is a follow up item which would eventually resolve this by creating a dedicated e2e test suite for releases instead of piggy-backing on the integration test suite:

https://github.com/Kong/kubernetes-ingress-controller/issues/1605